src/collection/collectionbackend.cpp

```diff
@@ -1381,11 +1381,23 @@ void CollectionBackend::UpdateLastPlayed(const QString &artist, const QString &a
 
   for (const Song &song : songs) {
     QSqlQuery q(db);
+    q.prepare(QString("SELECT lastplayed FROM %1 WHERE ROWID = :id").arg(songs_table_));
```

Comment:
We already got this in GetSongsBy() where Song::InitFromQuery is used. Just compare with song.lastplayed()

Reply:
Like this?

Reply:
Yes but continue instead of return, and remove the commented debug.
The code style should be consistent with the rest, space between ) and {. We also use 2 spaces for indent, not 4.

Reply:
OK, I will do it. If I change return to continue, I need to invert the comparison or negate it with ! , right?

Sorry, it is my first time with QT and C++!

Reply:
No. With continue we skip the updating SQL part and continue to the next song, ie.: another song on a different album.

Reply:
https://www.w3schools.com/cpp/cpp_break.asp